scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,501 @@
+    local petLvl = pet:getMainLvl()
+    local spellFamily = xi.magic.spellFamily.CURE
+    if
+        petLvl >= 16 and
+        math.random(100) <= 2 * xi.summon.getSummoningSkillOverCap(pet)
+    then
+        spellFamily = xi.magic.spellFamily.CURAGA
+    end
+
+    for _, spellData in ipairs(xi.pets.avatar.lightSpiritCures[spellFamily]) do
+        if petLvl >= spellData.level then
+            return spellData.spell
+        end
+    end
+
+    return 0
+end
+
+xi.pets.avatar.getLightSpiritSpell = function(pet)
+    -- returns the spirit's preferred target based on positioning
+    local master = pet:getMaster()
+
+    if not master then
+        return
+    end
+
+    local posTarget = nil
+    local posSpellID = 0
+    local buffTarget = nil
+    local buffSpellID = 0
+    local cureTarget = nil
+    local spellID = 0
+
+    if not pet:isEngaged() then
+        pet:lookAt(master:getPos())
+    else
+        -- don't do logic to buff anyone if pet is engaged. buffTarget will be reset below
+        buffTarget = master
+    end
+
+    local distance  = pet:checkDistance(master) -- starts as distance to master, updated to be distance to the posTarget
+    local hpp       = 0
+    local alliance  = master:getAlliance()
+    local party     = master:getParty()
+
+    for _, member in pairs(alliance) do
+        local tempHPP = member:getHPP()
+        if
+            pet:checkDistance(member) < 20 and
+            (cureTarget == nil or
+            tempHPP < hpp)
+        then
+            -- tiered chance of healing based on how low they are
+            -- flip a coin for every 25% hp the player has missing
+            for hpColor = 1, 3 do
+                if
+                    tempHPP < 25 * hpColor and
+                    math.random(100) < 50
+                then
+                    hpp = tempHPP
+                    cureTarget = member
+
+                    break
+                end
+            end
+        end
+    end
+
+    if cureTarget then
+        spellID = xi.pets.avatar.getLightSpiritCure(pet)
+
+        return spellID, cureTarget
+    else
+        local lightSpiritBuffs = xi.pets.avatar.getLightSpiritBuffs(pet)
+
+        for _, member in pairs(party) do
+            local tempDistance = pet:checkDistance(member)
+            if
+                not member:hasStatusEffect(xi.effect.INVISIBLE) and
+                (tempDistance <= distance or
+                not buffTarget)
+            then
+                -- Light Spirit will overwrite lower tiered Protect and Shell spells but will not overwrite Haste or Regen
+                local tempSpellID = 0
+
+                for _, spellData in ipairs(lightSpiritBuffs) do
+                    if
+                        not member:getStatusEffect(spellData.effect) or
+                        member:getStatusEffect(spellData.effect):getPower() < spellData.power
+                    then
+                        -- exit loop once a buff is found to be applied
```

Comment:
aaand if there is none to apply?

Reply:
then `tempSpellID` is left 0 and the function exits without setting a spellID